api/v1beta1/gcpmachine_webhook.go

```diff
@@ -117,3 +121,32 @@ func validateConfidentialCompute(spec GCPMachineSpec) error {
 	}
 	return nil
 }
+
+func checkKeyType(key *CustomerEncryptionKey) error {
+	if key.KeyType == CustomerManagedKey &&
+		(key.ManagedKey == nil || key.SuppliedKey != nil) {
+		return fmt.Errorf("CustomerEncryptionKey KeyType of Managed requires only ManagedKey to be set")
+	}
+	if key.KeyType == CustomerSuppliedKey &&
+		(key.SuppliedKey == nil || key.ManagedKey != nil) {
+		return fmt.Errorf("CustomerEncryptionKey KeyType of Supplied requires only SuppliedKey to be set")
+	}
+	return nil
```

Comment:
Want to handle KeyType being something other than the other two values? Maybe a switch with a default case of unknown value would be better?

Reply:
Good idea. Changed.